generators/javascript/text.js

```diff
@@ -38,7 +38,7 @@ Blockly.JavaScript['text'] = function(block) {
 Blockly.JavaScript['text_join'] = function(block) {
   // Create a string made up of any number of elements of any type.
   var code;
-  if (block.itemCount_ == 0) {
+  if (block.itemCount_ === 0) {
```

Comment:
I understand that your linter is scared of '0' == 0 == false == undefined == null.  But this linter does not appear to be very intelligent.  It doesn't seem to be aware of the JSDoc type system which defines the allowed types of each parameter, property and variable.  Google's JavaScript linter constructs a map of every definition and produces much smarter warnings.  This means that in the 42 times when we actually use ===, it serves as an obvious flag that interesting type checking is happening.

Besides, in this example (along with many others), if block.itemCount_ were undefined, then we'd want it to behave as though it were 0.  Otherwise, execution would lead to a loop where we would try to count from 0 to undefined.

Let's not rewrite the codebase just to reassure a mentally handicapped linter.  :)